File: clomosy_ai_trainer.js
const fs = require('fs');

const BASE_URL = "https://www.docs.clomosy.com/api.php";
const OUTPUT_MD = "TRObject_AI_KnowledgeBase.md";
const OUTPUT_JSON = "TRObject_AI_KnowledgeBase.json";

async function delay(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
}

async function getAllPages() {
    console.log("Sayfa listesi aliniyor...");
    let pages = [];
    let apfrom = null;
    
    while (true) {
        const url = new URL(BASE_URL);
        url.searchParams.append("action", "query");
        url.searchParams.append("list", "allpages");
        url.searchParams.append("aplimit", "max");
        url.searchParams.append("format", "json");
        
        if (apfrom) {
            url.searchParams.append("apcontinue", apfrom);
        }
        
        const response = await fetch(url.toString());
        const data = await response.json();
        
        const pageList = data.query?.allpages || [];
        for (const p of pageList) {
            pages.push(p.title);
        }
        
        if (data.continue?.apcontinue) {
            apfrom = data.continue.apcontinue;
        } else {
            break;
        }
    }
    
    console.log(`Toplam ${pages.length} sayfa bulundu.`);
    return pages;
}

async function fetchPageContent(title) {
    const url = new URL(BASE_URL);
    url.searchParams.append("action", "query");
    url.searchParams.append("prop", "revisions");
    url.searchParams.append("rvprop", "content");
    url.searchParams.append("titles", title);
    url.searchParams.append("format", "json");
    
    try {
        const response = await fetch(url.toString());
        const data = await response.json();
        const pages = data.query?.pages || {};
        
        for (const pageId in pages) {
            if (pages[pageId].revisions && pages[pageId].revisions.length > 0) {
                return pages[pageId].revisions[0]["*"];
            }
        }
    } catch (e) {
        console.error("Hata:", title, e);
    }
    return null;
}

async function main() {
    const pages = await getAllPages();
    
    let all_knowledge = [];
    let md_content = "# Clomosy TRObject Yapay Zeka Bilgi Bankasi\n\nBu dokuman otomatik olarak docs.clomosy.com adresinden derlenmistir.\n\n";
    
    for (let i = 0; i < pages.length; i++) {
        const title = pages[i];
        console.log(`[${i+1}/${pages.length}] Icerik cekiliyor: ${title}`);
        const content = await fetchPageContent(title);
        
        if (content) {
            all_knowledge.push({
                title: title,
                content: content
            });
            
            md_content += `## ${title}\n\n`;
            md_content += content + "\n\n---\n\n";
        }
        
        await delay(100); // sunucuya yuklenmemek icin bekle
    }
    
    console.log(`Veriler disari aktariliyor: ${OUTPUT_JSON}`);
    fs.writeFileSync(OUTPUT_JSON, JSON.stringify(all_knowledge, null, 2), "utf8");
    
    console.log(`Veriler disari aktariliyor: ${OUTPUT_MD}`);
    fs.writeFileSync(OUTPUT_MD, md_content, "utf8");
    
    console.log("Islem tamamlandi!");
}

main().catch(console.error);

File: _scripts/clomosy_ai_trainer.py
import requests
import json
import time
import os

BASE_URL = "https://www.docs.clomosy.com/api.php"
OUTPUT_MD = "TRObject_AI_KnowledgeBase.md"
OUTPUT_JSON = "TRObject_AI_KnowledgeBase.json"

def get_all_pages():
    print("Sayfa listesi aliniyor...")
    pages = []
    apfrom = None
    
    while True:
        params = {
            "action": "query",
            "list": "allpages",
            "aplimit": "max",
            "format": "json"
        }
        if apfrom:
            params["apcontinue"] = apfrom
            
        response = requests.get(BASE_URL, params=params)
        data = response.json()
        
        for p in data.get("query", {}).get("allpages", []):
            pages.append(p["title"])
            
        if "continue" in data:
            apfrom = data["continue"]["apcontinue"]
        else:
            break
            
    print(f"Toplam {len(pages)} sayfa bulundu.")
    return pages

def fetch_page_content(title):
    params = {
        "action": "query",
        "prop": "revisions",
        "rvprop": "content",
        "titles": title,
        "format": "json"
    }
    
    response = requests.get(BASE_URL, params=params)
    data = response.json()
    
    pages = data.get("query", {}).get("pages", {})
    for page_id, page_data in pages.items():
        if "revisions" in page_data:
            return page_data["revisions"][0]["*"]
    return None

def main():
    pages = get_all_pages()
    
    all_knowledge = []
    md_content = "# Clomosy TRObject Yapay Zeka Bilgi Bankasi\n\nBu dokuman otomatik olarak docs.clomosy.com adresinden derlenmistir.\n\n"
    
    for idx, title in enumerate(pages):
        print(f"[{idx+1}/{len(pages)}] Icerik cekiliyor: {title}")
        content = fetch_page_content(title)
        
        if content:
            all_knowledge.append({
                "title": title,
                "content": content
            })
            
            md_content += f"## {title}\n\n"
            md_content += content
            md_content += "\n\n---\n\n"
            
        # Sunucuya yuklenmemek icin bekle
        time.sleep(0.1)
        
    print(f"Veriler disari aktariliyor: {OUTPUT_JSON}")
    with open(OUTPUT_JSON, "w", encoding="utf-8") as f:
        json.dump(all_knowledge, f, ensure_ascii=False, indent=2)
        
    print(f"Veriler disari aktariliyor: {OUTPUT_MD}")
    with open(OUTPUT_MD, "w", encoding="utf-8") as f:
        f.write(md_content)
        
    print("Islem tamamlandi!")

if __name__ == "__main__":
    main()
